presto-main/src/main/java/com/facebook/presto/operator/scalar/QuantileDigestFunctions.java

```diff
@@ -101,6 +101,37 @@ public static Block valuesAtQuantilesBigint(@SqlType("qdigest(bigint)") Slice in
         return output.build();
     }
 
+    @ScalarFunction("scale_qdigest")
+    @Description("Scale a quantile digest according to a new weight")
+    @SqlType("qdigest(double)")
+    public static Slice scaleQuantileDigesDouble(@SqlType("qdigest(double)") Slice input, @SqlType(StandardTypes.DOUBLE) double scale)
+    {
+        return scaleQuantileDiges(input, scale);
+    }
+
+    @ScalarFunction("scale_qdigest")
+    @Description("Scale a quantile digest according to a new weight")
+    @SqlType("qdigest(real)")
+    public static Slice scaleQuantileDigesReal(@SqlType("qdigest(real)") Slice input, @SqlType(StandardTypes.DOUBLE) double scale)
+    {
+        return scaleQuantileDiges(input, scale);
+    }
+
+    @ScalarFunction("scale_qdigest")
+    @Description("Scale a quantile digest according to a new weight")
+    @SqlType("qdigest(bigint)")
+    public static Slice scaleQuantileDigesBigint(@SqlType("qdigest(bigint)") Slice input, @SqlType(StandardTypes.DOUBLE) double scale)
+    {
+        return scaleQuantileDiges(input, scale);
+    }
+
+    private static Slice scaleQuantileDiges(Slice input, double scale)
+    {
+        QuantileDigest digest = new QuantileDigest(input);
```

Comment:
Let's validate that the scale is > 0 and throw `new PrestoException(INVALID_FUNCTION_ARGUMENT, ....)` otherwise.

Reply:
This is marked as resolved, however I don't see this implemented.

Reply:
I mess up this when doting the conflict resolve. added it back.